src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -1,15 +1,127 @@
 package TypewiseAlert;
 
-import static org.junit.Assert.assertTrue;
-
+import org.junit.Before;
 import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
 
 public class TypewiseAlertTest 
 {
+    private EmailDispatcher emailDispatcher;
+    private DeviceController deviceController;
+    private TypewiseAlert typewiseAlert;
+    private TemperatureBreachClassifier temperatureBreachClassifier;
+
+    @Before
+    public void setup() {
+        emailDispatcher = new EmailDispatcher();
+        deviceController = new DeviceController();
+        typewiseAlert = new TypewiseAlert();
+        temperatureBreachClassifier = new TemperatureBreachClassifier();
+    }
+
     @Test
     public void infersBreachAsPerLimits()
     {
-      assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
+        assertSame(temperatureBreachClassifier.inferBreach(12, 20, 30), BreachType.TOO_LOW);
+        assertSame(temperatureBreachClassifier.inferBreach(40, 20, 30), BreachType.TOO_HIGH);
+        assertSame(temperatureBreachClassifier.inferBreach(20, 20, 30), BreachType.NORMAL);
+    }
+
+    @Test
+    public void getLimitsTest() {
+        assertTrue(CoolingType.PASSIVE_COOLING.getLowTemp() == 0);
+        assertTrue(CoolingType.PASSIVE_COOLING.getHighTemp() == 35);
+        assertTrue(CoolingType.MED_ACTIVE_COOLING.getLowTemp() == 0);
+        assertTrue(CoolingType.MED_ACTIVE_COOLING.getHighTemp() == 40);
+        assertTrue(CoolingType.HI_ACTIVE_COOLING.getLowTemp() == 0);
+        assertTrue(CoolingType.HI_ACTIVE_COOLING.getHighTemp() == 45);
+    }
+
+    @Test
+    public void sendToEmailTargetTest() {
+        String responseMsg = emailDispatcher.dispatchBreachMsgToTarget(BreachType.TOO_LOW);
```

Comment:
the test case should check exactly what is expected return value. Not null  is a weak check. It can so happen that high message is returned whereas low message is expected.